brayns/common/utils/ImageUtils.h

```diff
@@ -0,0 +1,102 @@
+/* Copyright (c) 2015-2018, EPFL/Blue Brain Project
+ * All rights reserved. Do not distribute without permission.
+ *
+ * This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#pragma once
+
+#include <brayns/common/renderer/FrameBuffer.h>
+
+#ifdef BRAYNS_USE_FREEIMAGE
+#include <FreeImage.h>
+
+namespace brayns
+{
+namespace freeimage
+{
+struct ImageDeleter
+{
+    inline void operator()(FIBITMAP* image)
+    {
+        if (image)
+            FreeImage_Unload(image);
+    }
+};
+
+struct MemoryDeleter
+{
+    inline void operator()(FIMEMORY* memory)
+    {
+        if (memory)
+            FreeImage_CloseMemory(memory);
+    }
+};
+
+using ImagePtr = std::unique_ptr<FIBITMAP, ImageDeleter>;
+using MemoryPtr = std::unique_ptr<FIMEMORY, MemoryDeleter>;
+
+template <class T>
+inline void INPLACESWAP(T& a, T& b)
+{
+    a ^= b;
+    b ^= a;
+    a ^= b;
+}
+
+// https://github.com/imazen/freeimage/blob/master/Source/FreeImage/Conversion.cpp#L56
+bool SwapRedBlue32(FIBITMAP* dib)
+{
+    if (FreeImage_GetImageType(dib) != FIT_BITMAP)
+        return false;
+
+    const unsigned bytesperpixel = FreeImage_GetBPP(dib) / 8;
+    if (bytesperpixel > 4 || bytesperpixel < 3)
+        return false;
+
+    const unsigned height = FreeImage_GetHeight(dib);
+    const unsigned pitch = FreeImage_GetPitch(dib);
+    const unsigned lineSize = FreeImage_GetLine(dib);
+
+    BYTE* line = FreeImage_GetBits(dib);
+    for (unsigned y = 0; y < height; ++y, line += pitch)
+        for (BYTE* pixel = line; pixel < line + lineSize;
+             pixel += bytesperpixel)
+            INPLACESWAP(pixel[0], pixel[2]);
+    return true;
+}
+
+ImagePtr getImageFromFrameBuffer(FrameBuffer& frameBuffer)
```

Comment:
getImageFromFrameBuffer and SwapRedBlue32 should go into ImageUtils.cpp. You can get compilation errors about double definitions if you try using them in two places, unless the function are inlined, but I don't see a good reason to inline them.